docs/source/cpp/io.rst

```diff
@@ -0,0 +1,86 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: cpp
+.. highlight:: cpp
+
+==============================
+Input / output and filesystems
+==============================
+
+Arrow provides a range of C++ interfaces abstracting the concrete details
+of input / output operations.  They operate on streams of untyped binary data.
+Those abstractions are used for various purposes such as reading CSV or
+Parquet data, transmitting IPC streams...
```

Comment:
Is this `...` a note to self to fill this in, or is it supposed to be there?

Reply:
It's just a way of writing "etc.". Is it incorrect?

Reply:
```suggestion
Parquet data, transmitting IPC streams, and more.
```